src/qgraphicsmozview_p.cpp

```diff
@@ -295,10 +303,7 @@ void QGraphicsMozViewPrivate::OnLoadStarted(const char* aLocation)
 {
     Q_UNUSED(aLocation);
 
-    if (mIsPainted) {
-        mIsPainted = false;
-        mViewIface->firstPaint(-1, -1);
-    }
+    ResetPainted();
```

Comment:
Oh, we have mIsPainted variable, which is endup in wrong state.
Let me check if something like "pagehide" https://github.com/tmeshkova/gecko-dev/blob/embedlite/embedding/embedlite/utils/WebBrowserChrome.cpp#L421 help to reset that state in the middle of navigation between docshell's.